framework/include/base/Assembly.h

```diff
@@ -189,6 +189,16 @@ class Assembly
   const Elem *& elem() { return _current_elem; }
 
   /**
+   * Return the current subdomain ID
+   */
+  const SubdomainID & currentSubdomainID() { return _current_subdomain_id; }
```

Comment:
Probably should be marked `const` so it can be called from a const object later.